src/IRMatch.h

```diff
@@ -1483,7 +1483,48 @@ struct Intrin {
         return Expr();
     }
 
-    constexpr static bool foldable = false;
+    constexpr static bool foldable = true;
+
+    HALIDE_ALWAYS_INLINE void make_folded_const(halide_scalar_value_t &val, halide_type_t &ty, MatcherState &state) const noexcept {
+        halide_scalar_value_t arg1;
+        // Assuming the args have the same type as the intrinsic is incorrect in
+        // general. But for the intrinsics we can fold (just shifts), the LHS
+        // has the same type as the intrinsic, and we can always treat the RHS
+        // as a signed int, because we're using 64 bits for it. (TODO: Dillon,
```

Comment:
Yes, I think this is OK. Constants will be simplified to unsigned shifts if needed.